### ISSUE TITLE
cxl-marketing-nav: context menu should not close on item right-click

### ISSUE THREAD

**Issue by lkraav** [conversionxl/aybolit#41]
People might want to share a context menu item link, but the element closes immediately on right-click.

Are we maybe doing too much or too little at https://github.com/conversionxl/aybolit/blob/cxl-ui-0.1.9/packages/cxl-ui/src/components/cxl-marketing-nav.js#L245

**Comment by freudFlintstone** [comment]
Couldn't quite solve this, but this is the current status:

- it's a known issue in vaadin: https://github.com/vaadin/vaadin-context-menu/issues/161
- item-selected is not the right event for this
- Holding the shift key makes this work, because vaadin-context-menu has a _contextmenu_ event listener that closes the menu if the event otherwise:
https://github.com/vaadin/vaadin-context-menu/blob/48507118abd95baaae29892e63c4929618450cf0/src/vaadin-context-menu.html#L741-L748
`_onGlobalContextMenu(e) {
    if (!e.shiftKey) {
      e.preventDefault();
      this.close();
    }
  }
`
-  however, trying to listen to _'contextmenu'_ or the custom event '_vaadin-contextmenu' (defined in https://github.com/vaadin/vaadin-context-menu/blob/master/src/vaadin-contextmenu-event.html) also won't work.

I'm quite conifdent there's a way to solve this by using the contextmenu, but I'd like some input

**Comment by lkraav** [comment]
> Holding the shift key makes this work

Indeed, great find. This gives us a lead to chase - how can we make it behave the same without Shift key.

Sub-classing is one clear option, but I'd like to avoid bringing in a whole component into our library for this.

Can we add another event handler on top to stop the close?

**Comment by freudFlintstone** [comment]
I tried a few things following the DOM hierarchy, but it the context-menu-event contextmenu, based on polymer.gestures, is never prevented. I'd have to look at polymer.gestures to understand how.

I was actually able to capture the event, by adding a listener to the event that trigger close on the overlay, and on each item in the context menu. But both of them happen after the menu closes. SO we might have to do something about that `_onGlobalContextMenu` anyway

**Comment by lkraav** [comment]
I commented on the Vaadin repo issue. Maybe there will be a response tomorrow of some kind.

**Comment by lkraav** [comment]
2023: issue still exists.

Upstream update https://github.com/vaadin/web-components/issues/4769
